src/App/Containers/Posts/PostContainer.js

```diff
@@ -4,38 +4,67 @@ import CardModule from '../../components/Card/index.card'
 import SearchModule from '../../components/Search/index.search'
 
 class PostContainer extends Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      postData: [],
-      slicedData: []
-    }
+  // constructor(props) {
+  //   super(props)
+  //   this.state = {
+  //     postData: [],
+  //     result: '',
+  //     searchString: '',
+  //     slicedData: []
+  //   }
+  // }
+
+  state = {
+    postData: [],
+    result: '',
+    searchString: '',
+    slicedData: []
   }
+  // };
 
   componentDidMount() {
-    fetch('https://jsonplaceholder.typicode.com/posts')
+    // this.setState({ searchString: '' })
+    this.fetchData()
+  }
+
+  fetchData = (searchString) => {
+    let url
+    url = 'https://jsonplaceholder.typicode.com/posts'
+
+    if (searchString) {
+      url = `${url}/${searchString}`
+    }
+    fetch(url)
```

Comment:
Can able to use fetch as a general helper function and reuse it in different components.